confluence/server/scio_search/src/main/java/com/askscio/atlassian_plugins/confluence/impl/Utils.java

```diff
@@ -0,0 +1,68 @@
+package com.askscio.atlassian_plugins.confluence.impl;
+
+import static com.askscio.atlassian_plugins.confluence.impl.MyPluginComponentImpl.PLUGIN_STATUS_KEY;
+
+import com.atlassian.sal.api.pluginsettings.PluginSettings;
+import com.atlassian.sal.api.user.UserManager;
+import com.atlassian.sal.api.user.UserProfile;
+import com.google.gson.Gson;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import org.codehaus.jackson.annotate.JsonAutoDetect;
+import org.codehaus.jackson.annotate.JsonAutoDetect.Visibility;
+
+public class Utils {
+
+  private static final Gson gson = new Gson();
+
+  public static boolean isCurrentUserAdmin(UserManager userManager) {
+    final UserProfile profile = userManager.getRemoteUser();
+    return profile != null && userManager.isSystemAdmin(profile.getUserKey());
+  }
+
+  public static void validateUserIsAdmin(UserManager userManager) {
+    if (!isCurrentUserAdmin(userManager)) {
+      throw new UnauthorizedException("Unauthorized");
+    }
+  }
+
+  public static PluginStatus getPluginStatus(PluginSettings pluginSettings) {
+    if (pluginSettings.get(PLUGIN_STATUS_KEY) == null) {
+      return new PluginStatus();
+    }
+    return gson.fromJson((String) pluginSettings.get(PLUGIN_STATUS_KEY), PluginStatus.class);
+  }
+
+  public static void updatePluginStatus(PluginSettings pluginSettings, int responseCode) {
```

Comment:
Any idea how costly this is? I might be inclined to just update it once per minute or so. Every single page view seems excessive.

Reply:
Makes sense - added a cache and set the write frequency to 1 minute